demos/tests/run_tests.py

```diff
@@ -143,13 +157,29 @@ def option_to_args(key, value):
                     ' '.join(shlex.quote(str(arg)) for arg in case_args))
                 print(flush=True)
 
+                execution_time = -1
+                device = ""
+                pipeline = []
+                for key, value in test_case.options.items():
+                    match = re.search(r'-d.*', key)
+                    if match and device == "":
+                        device = value
+                    match = re.search(r'-m.*', key)
+                    if match:
+                        model_path = option_to_args(key, value)[1]
+                        model = re.search(r'.*/(models/.*).xml', model_path)
+                        pipeline.append(model.group(1) if model else option_to_args(key, value)[1])
+                pipeline.sort()
                 try:
+                    start_time = timeit.default_timer()
                     subprocess.check_output(fixed_args + case_args,
                         stderr=subprocess.STDOUT, universal_newlines=True)
+                    execution_time = timeit.default_timer() - start_time
                 except subprocess.CalledProcessError as e:
                     print(e.output)
                     print('Exit code:', e.returncode)
                     num_failures += 1
+                parse_result(demo.full_name, device, pipeline, execution_time)
```

Comment:
This will record a nonsensical time if the demo fails to run. IMO, you shouldn't record results for unsuccessful cases.